tests/terminator_tests/improvement_tests/test_preprocessing.py

```diff
@@ -0,0 +1,236 @@
+        assert np.log(dist_before.low) == dist_after.low
+        assert np.log(dist_before.high) == dist_after.high
+
+
+@pytest.mark.parametrize("direction", (StudyDirection.MINIMIZE, StudyDirection.MAXIMIZE))
+def test_select_top_trials(direction: StudyDirection) -> None:
+    values = [1, 0, 0, 2, 1]
+    trials_before = [create_trial(value=v) for v in values]
+
+    values_in_order = sorted(values, reverse=(direction == StudyDirection.MAXIMIZE))
+
+    # Scenario: `n_trials` * `top_trials_ratio` is less than `min_n_trials`.
+    top_trials_ratio = 0.4
+    min_n_trials = 3
+    p = _preprocessing.SelectTopTrials(
+        top_trials_ratio=top_trials_ratio,
+        min_n_trials=min_n_trials,
+    )
+    trials_after = p.apply(trials_before, direction)
+
+    n_trials_expected = min_n_trials
+    assert len(trials_after) == n_trials_expected
+    assert _get_trial_values(trials_after) == values_in_order[:n_trials_expected]
+
+    # Scenario: `n_trials` * `top_trials_ratio` is greater than `min_n_trials`.
+    top_trials_ratio = 0.8
+    min_n_trials = 3
+    p = _preprocessing.SelectTopTrials(
+        top_trials_ratio=top_trials_ratio,
+        min_n_trials=min_n_trials,
+    )
+    trials_after = p.apply(trials_before, direction)
+
+    n_trials_expected = int(len(values) * top_trials_ratio)
+    assert len(trials_after) == n_trials_expected
+    assert _get_trial_values(trials_after) == values_in_order[:n_trials_expected]
+
+
+@pytest.mark.parametrize("direction", (StudyDirection.MINIMIZE, StudyDirection.MAXIMIZE))
+def test_to_minimize(direction: StudyDirection) -> None:
+    values = [1, 0, 0, 2, 1]
+    trials_before = [create_trial(value=v) for v in values]
+
+    expected_values = values[:]
+    if direction == StudyDirection.MAXIMIZE:
+        expected_values = [-1 * v for v in expected_values]
+
+    p = _preprocessing.ToMinimize()
+    trials_after = p.apply(trials_before, direction)
+    assert _get_trial_values(trials_after) == expected_values
+
+
+@pytest.mark.parametrize("direction", (StudyDirection.MINIMIZE, StudyDirection.MAXIMIZE))
+def test_one_to_hot(direction: StudyDirection) -> None:
+    # Scenario: categorical parameters are mapped to the float distribution.
+    trials_before = [
+        create_trial(
+            params={"categorical": 0},
+            distributions={"categorical": CategoricalDistribution((0, 1))},
+            value=1.0,
+        ),
+        create_trial(
+            params={"categorical": 1},
+            distributions={"categorical": CategoricalDistribution((0, 1))},
+            value=1.0,
+        ),
+    ]
+    p = _preprocessing.OneToHot()
+    trials_after = p.apply(trials_before, direction)
+
+    for t in trials_after:
+        # Distributions are expected to be mapped to `FloatDistribution(0, 1)` for each choice.
+        assert len(t.distributions) == 2
+        for d in t.distributions.values():
+            assert d == FloatDistribution(0, 1)
```

Comment:
I think these assertions do not check the names of the distributions. How about checking them like `trials_after[0].params`?
```suggestion
        assert t.distributions == {
            "i0_categorical": FloatDistribution(0, 1),
            "i1_categorical": FloatDistribution(0, 1),
        }
```